### DISCUSSION TITLE
Removing elements from a JSON object

### DISCUSSION THREAD

**Discussion by andysCaplin** [danielaparker/jsoncons#347] [Q&A] [answered]
Hi, I'm creating a JSON object to hold a list of subjects each of which has an associated list of fields - the object ends up looking like shown below

```
{
  "objects": {
    "/SYSTEM/USER": {
      "fields": {
        "ServiceState": "UP"
      }
    },
    "/SYSTEM/INFO": {
      "fields": {
        "SourceID": "0"
      }
    }
  }
}
```

I create the elements using the [] notation

```
status_page.["objects"][subject]["fields"][name] = value;
```

I want to be able to delete an object with a given subject - e.g. the entry for "/SYSTEM/USER". There doesn't seem to be an equivalent notation to completely delete it. I tried

```
status_page.["objects"][subject] = NULL;
```

That sets the contents to 0 but leaves the element for the subject in.

I tried using json_pointer::remove() but it expects '/' as a separator, but my subjects already have '/'s - i.e. I can't use "objects//SYSTEM/USER".

Is there any way of achieving this without turning it into an array and using the method you described in https://github.com/danielaparker/jsoncons/discussions/289 ?

**Comment by danielaparker** [comment, marked as answer]
To remove with a JSON Pointer, all occurrences of `~` and `/` in the member name must be escaped with `~0` and `~1` respectively, so you'd need to use
```cpp
json val = json::parse(data);
jsonpointer::remove(val, "/objects/~1SYSTEM~1USER");
```
Alternatively, you could use the `json_pointer` class to do the escaping for you,
```
jsonpointer::json_pointer ptr;
ptr /= "objects";
ptr /= "/SYSTEM/USER";
jsonpointer::remove(val, ptr);
``` 
But the simplest way would be to use
```
val["objects"].erase("/SYSTEM/USER");
```

**Comment by andysCaplin** [comment]
Thank you. That worked perfectly - I used the erase.
